src/tss2-tcti/tcti-spidev.c

```diff
@@ -0,0 +1,202 @@
+    PLATFORM_USERDATA *platform_data = (PLATFORM_USERDATA *) user_data;
+
+    tr.cs_change = 1;
+    tr.len = cnt;
+    tr.tx_buf = (unsigned long) data_out;
+    tr.rx_buf = (unsigned long) data_in;
+	int ret = ioctl(platform_data->fd, SPI_IOC_MESSAGE(1), &tr);
+    if (ret < 0) {
+        LOG_ERROR("SPI acquire failed: %s", strerror(errno));
+        return TSS2_TCTI_RC_IO_ERROR;
+    }
+    LOGBLOB_DEBUG(data_in, cnt, "Received data over ioctl:");
+    return TSS2_RC_SUCCESS;
+}
+
+TSS2_RC
+platform_sleep_ms (void *user_data, int32_t milliseconds)
+{
+    (void) user_data;
+    struct timeval tv = {milliseconds/1000, (milliseconds%1000)*1000};
+    select (0, NULL, NULL, NULL, &tv);
+
+    return TSS2_RC_SUCCESS;
+}
+
+TSS2_RC
+platform_start_timeout (void *user_data, int32_t milliseconds)
+{
+    PLATFORM_USERDATA *platform_data = (PLATFORM_USERDATA *) user_data;
+
+    memset (&platform_data->timeout, 0, sizeof (struct timeval));
+
+    if (gettimeofday (&platform_data->timeout, NULL)) {
+        LOG_ERROR ("getimeofday failed with errno: %d.", errno);
+        return TSS2_TCTI_RC_GENERAL_FAILURE;
+    }
+
+    platform_data->timeout.tv_sec += (milliseconds/1000);
+    platform_data->timeout.tv_usec += (milliseconds%1000)*1000;
+    if (platform_data->timeout.tv_usec > 999999) {
+        platform_data->timeout.tv_sec++;
+        platform_data->timeout.tv_usec -= 1000000;
+    }
+
+    return TSS2_RC_SUCCESS;
+}
+
+TSS2_RC
+platform_timeout_expired (void *user_data, bool *is_timeout_expired)
+{
+    PLATFORM_USERDATA *platform_data = (PLATFORM_USERDATA *) user_data;
+
+    struct timeval now;
+    if (gettimeofday (&now, NULL)) {
+        LOG_ERROR ("getimeofday failed with errno: %d.", errno);
+        return TSS2_TCTI_RC_GENERAL_FAILURE;
+    }
+
+    if (now.tv_sec > platform_data->timeout.tv_sec) {
+        *is_timeout_expired = true;
+    } else if ((now.tv_sec == platform_data->timeout.tv_sec)
+               && (now.tv_usec > platform_data->timeout.tv_usec)) {
+        *is_timeout_expired = true;
+    } else {
+        *is_timeout_expired = false;
+    }
+
+    return TSS2_RC_SUCCESS;
+}
+
+void
+platform_finalize(void *user_data)
+{
+    PLATFORM_USERDATA *platform_data = (PLATFORM_USERDATA *) user_data;
+    close(platform_data->fd);
+    free(platform_data);
+}
+
+TSS2_RC
+Tss2_Tcti_Spidev_Init (TSS2_TCTI_CONTEXT* tcti_context, size_t* size, const char* config)
+{
+    TSS2_TCTI_SPI_HELPER_PLATFORM platform = {0};
+
+    if (!config || config[0] == '\0')
+        config = "/dev/spidev0.1";
+
+    /* Check if context size is requested */
+    if (tcti_context == NULL) {
+        return Tss2_Tcti_Spi_Helper_Init (NULL, size, NULL);
+    }
+
+    /* Create required platform user data */
+    PLATFORM_USERDATA *platform_data = calloc (1, sizeof (PLATFORM_USERDATA));
+    if (platform_data == NULL) {
+        return TSS2_BASE_RC_MEMORY;
+    }
+
+    platform_data->fd = open(config, O_RDWR);
```

Comment:
The TCTI config is documented nowhere. We probably want to add a man page. We might want to add it to `doc/tcti.md`, as well.

Reply:
Added manpage and section to tcti.md.